### PR TITLE
Populate pDNS fallback IPs in BIND configuration

### PR DESCRIPTION
**Pull request by emileswarts**
# What
Populate pDNS IPs for queries that don't match a specific zone configuration.
These IPs of pDNS are dynamic and injected into the configuration from
environment variables.

# Why
When no zones match (such as for public internet queries) we want to
forward these requests to the NCSC pDNS servers to deal with.

### CODE REVIEW

File: app/lib/use_cases/generate_bind_config.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ def initialize(zones: [])
   end
 
   def execute
+    pdns_ips = JSON.parse(ENV.fetch("PDNS_IPS")).join(";\n")
```
Comment (jivdhaliwal) [inline app/lib/use_cases/generate_bind_config.rb:7]: What if pass in the PDNS IPs as another param in the initialiser? That way the use case doesn't need to have any knowledge of the ENV

**Review by jivdhaliwal — APPROVED** [review]

**Review by jbevan4 — APPROVED** [review]

File: app/lib/use_cases/generate_bind_config.rb (hunk 1 above)
Comment (jbevan4) [reply app/lib/use_cases/generate_bind_config.rb:7]: I'm struggling to see the value in this. If we move this into the initialiser, the controller which invokes this use-case then becomes environment aware?

Comment (jivdhaliwal) [reply app/lib/use_cases/generate_bind_config.rb:7]: The value being that the use case has as little knowledge of the system as possible, and only deals with the values it is initialised with. The ENV is global to the app so the controller is already aware of it / has it available to pass in.

Comment (jbevan4) [reply app/lib/use_cases/generate_bind_config.rb:7]: Cool, guess I'm just writing the test wrong as it keeps failing

**Pull request merged**